src/connection/conn_string.rs

```diff
@@ -0,0 +1,430 @@
+//! Connection string parser
+
+use crate::*;
+use std::str::FromStr;
+use url::Url;
+
+pub struct ConnStringSettings {
+    pub user: Option<String>,
+    pub pass: Option<String>,
+    pub host: Option<String>,
+    pub port: Option<u16>,
+    pub db_name: String,
+    pub charset: Option<Charset>,
+    pub dialect: Option<Dialect>,
+    pub lib_path: Option<String>,
+    pub stmt_cache_size: Option<usize>,
+}
+
+/// Parse the connection string.
+///
+/// Basic string sintax: `firebird://{user}:{pass}@{host}:{port}/{db_name}?{options}`
+pub fn parse(sconn: &str) -> Result<ConnStringSettings, FbError> {
```

Comment:
Some questions:
- It follows IETF RFC [3986](https://tools.ietf.org/html/rfc3986) ? (Eg.: RFC [7869](https://tools.ietf.org/html/rfc7869))
- It's worth any provision for handling distinct drivers/implementations/modes like [SQLAlchemy](https://docs.sqlalchemy.org/en/14/dialects/mysql.html#mariadb-support) does?

Reply:
>     * It follows IETF RFC [3986](https://tools.ietf.org/html/rfc3986) ? (Eg.: RFC [7869](https://tools.ietf.org/html/rfc7869))
 The crate used for parsing follows [this](https://url.spec.whatwg.org/) spec. It seems to be similar, but not identical to RFC 3986 and RFC 3987.

Reply:
> It follows IETF RFC [3986](https://tools.ietf.org/html/rfc3986) ? (Eg.: RFC [7869](https://tools.ietf.org/html/rfc7869))

Yes, I think. Others databases use this same 'string layout'. Why?

Reply:
> It's worth any provision for handling distinct drivers/implementations/modes like [SQLAlchemy](https://docs.sqlalchemy.org/en/14/dialects/mysql.html#mariadb-support) does?

Sorry, I'm not following you. To choose between our two variations(pure rust/native) you must use the respective connection builder, like `builder_native()`. And to use the dynamic linked or dynamic loaded, you use the `?lib=` parameter.